### PR TITLE
Resolved #3147 where templates export did not work when using PHP 8.1

### LINKED ISSUE #3147: Template downloads failing after update from EE5 to EE6
I am seeing error messages post-update from EE 5.x.x to EE 6.4.6 AND 6.4.7 on multiple sites running on a Dreamhost server. Specifically, the template download function fails and shows a series of error messages and certain front end pages yeild a separate set of error messages. The problems go away if I revert back to PHP 7.4 from PHP 8.x. All add-ons are up to date.

~~~~~~~~~~~~~~~~~~~~~~~~~~~~
WHEN TRYING TO DOWNLOAD TEMPLATES (OPERATION FAILS COMPLETELY):

Deprecated
ZipArchive::open(): Using empty file as ZipArchive is deprecated
ee/ExpressionEngine/Controller/Design/AbstractDesign.php, line 331
Severity: E_DEPRECATED

Warning
Cannot modify header information - headers already sent by (output started at ee/legacy/core/Exceptions.php:120)
ee/legacy/helpers/download_helper.php, line 54
Severity: E_WARNING

WHEN LOADING CERTAIN PAGES ON FRONT END:

Deprecated
ctype_digit(): Argument of type bool will be interpreted as string in the future
ee/ExpressionEngine/Addons/channel/mod.channel.php, line 4187
Severity: E_DEPRECATED

Warning
Cannot modify header information - headers already sent by (output started at ee/legacy/core/Exceptions.php:120)
ee/ExpressionEngine/Boot/boot.common.php, line 475
Severity: E_WARNING

### PR DESCRIPTION
**Pull request by intoeetive**
Resolved #3147 where templates export did not work when using PHP 8.1

### CODE REVIEW

File: system/ee/ExpressionEngine/Controller/Design/AbstractDesign.php [hunk 1]
```diff
@@ -322,7 +322,7 @@ protected function exportTemplates()
         // Create the Zip Archive
         $zipfilename = tempnam(sys_get_temp_dir(), '');
         $zip = new ZipArchive();
-        if ($zip->open($zipfilename, ZipArchive::CREATE) !== true) {
+        if ($zip->open($zipfilename, ZipArchive::OVERWRITE) !== true) {
```
Comment (matthewjohns0n) [inline system/ee/ExpressionEngine/Controller/Design/AbstractDesign.php:325]: Would it be better to use `ZipArchive::CREATE | ZipArchive::OVERWRITE` in this instance?

Comment (intoeetive) [reply system/ee/ExpressionEngine/Controller/Design/AbstractDesign.php:325]: Why would be :: OVERWRITE not sufficient?

Comment (matthewjohns0n) [reply system/ee/ExpressionEngine/Controller/Design/AbstractDesign.php:325]: I was under the impression that `::OVERWRITE` only works when there is a file to overwrite, and that you needed to use `::CREATE` if you wanted to create a file. 

See: https://stackoverflow.com/questions/49164547/ziparchive-i-need-to-create-new-file-and-overwrite-existing

Comment (intoeetive) [reply system/ee/ExpressionEngine/Controller/Design/AbstractDesign.php:325]: We are creating file with `tempnam` here, so it's always there

Comment (matthewjohns0n) [reply system/ee/ExpressionEngine/Controller/Design/AbstractDesign.php:325]: Okay thats what I was missing. Sounds good!

**Review by matthewjohns0n — APPROVED** [review]

**Pull request merged**